engine/src/main/java/org/terasology/logic/ai/HierarchicalAISystem.java

```diff
@@ -125,17 +125,17 @@ private void loop(EntityRef entity, LocationComponent location,
 
             // if our AI is aggressive or hunter go and hunt player else run away
             // if wild
-            if (ai.aggressive) {
+            /*if (ai.aggressive) {
```

Comment:
I think that just commenting here is not the best solution. You could either
- try to fix the code (if possible)
- re-structure and organize the comments and TODOs into a proper task description and pseudocode solution. this should probably be an issue on its own, referring to the legacy code